contracts/token/ERC721/extensions/ERC721Enumerable.sol

```diff
@@ -24,6 +24,18 @@ abstract contract ERC721Enumerable is ERC721, IERC721Enumerable {
     // Mapping from token id to position in the allTokens array
     mapping(uint256 => uint256) private _allTokensIndex;
 
+    /**
+     * @dev An `owner`'s token query was out of bounds for `index`.
+     *
+     * NOTE: The owner being `address(0)` indicates a global out of bounds index.
+     */
+    error ERC721OutOfBoundsIndex(address owner, uint256 index);
```

Comment:
Same comment as `Paused()` and `InvalidArrayLength`: IMO `OutOfBoundsAccess(uint256 index)` should be a generic error without a domain.